rcon/squad_automod/automod.py

```diff
@@ -63,154 +65,151 @@ def is_time(times: List[datetime], interval_seconds: int):
+        return PunishStepState.WAIT
+
     if config.number_of_warning == 0:
         logger.debug("Warnings are disabled. number_of_warning is set to 0")
-        return PunishStepState.disabled
-
-    warnings = watch_status.warned.setdefault(player["name"], [])
+        return PunishStepState.DISABLED
 
     if not is_time(warnings, config.warning_interval_seconds):
-        logger.debug("Waiting to warn: %s in %s", player, squad_name)
-        return PunishStepState.wait
+        logger.debug("Waiting to warn: %s in %s", aplayer.short_repr(), squad_name)
+        return PunishStepState.WAIT
 
-    if len(warnings) < config.number_of_warning or config.number_of_warning == -1:
+    warning_count = len(warnings) - 1
+    if warning_count < config.number_of_warning or config.number_of_warning == -1:
         logger.info(
-            "Player: %s Level: %s Role: %s Will be warned (%s/%s)",
-            player["name"],
-            player["level"],
-            player["role"],
-            len(warnings),
+            "%s Will be warned (%s/%s)",
+            aplayer.short_repr(),
+            warning_count,
             num_or_inf(config.number_of_warning),
         )
         warnings.append(datetime.now())
-        return PunishStepState.apply
+        return PunishStepState.APPLY
 
     logger.info(
-        "Player: %s Level: %s Role: %s Max warnings reached (%s/%s). Moving on to punish.",
-        player["name"],
-        player["level"],
-        player["role"],
-        len(warnings),
+        "%s Max warnings reached (%s/%s). Moving on to punish.",
+        aplayer.short_repr(),
+        warning_count,
         config.number_of_warning,
     )
-    return PunishStepState.go_to_next_step
-
-
-def _get_team_count(team_view, team):
-    return sum(
-        len(s.get("players", [])) for s in team_view[team].get("squads", {}).values()
-    )
+    return PunishStepState.GO_TO_NEXT_STEP
 
 
 def should_punish_player(
     watch_status: WatchStatus,
     config: NoLeaderConfig,
     team_view,
-    squad_name,
+    squad_name: str,
     squad,
-    player,
+    aplayer: APlayer,
 ):
     if config.number_of_punish == 0:
         logger.debug("Punish is disabled")
-        return PunishStepState.disabled
+        return PunishStepState.DISABLED
 
     if (
         _get_team_count(team_view, "allies") + _get_team_count(team_view, "axis")
     ) < config.disable_punish_below_server_player_count:
         logger.debug("Server below min player count for punish")
-        return PunishStepState.wait
+        return PunishStepState.WAIT
 
     if len(squad["players"]) < config.min_squad_players_for_punish:
         logger.debug("Squad %s below min player count for punish", squad_name)
-        return PunishStepState.wait
+        return PunishStepState.WAIT
 
     if (
-        int(player["level"]) <= config.immuned_level_up_to
-        or player["role"] in config.immuned_roles
+        aplayer.lvl <= config.immuned_level_up_to
```

Comment:
I wouldn't remove the type case here. I'm not sure the playerinfo method is casting it and APlayer doesn't either